build/cluster.js

```diff
@@ -150,7 +124,7 @@ gulp.task('wait-for-heapster', function(doneFn) {
 
     // constantly query the heapster until it is properly running
     heapsterHealthCheck(function(result) {
-      if (result === runningState) {
+      if (!isNaN(result)) {
```

Comment:
How about you call-back this function only when heapster is up and running. And do the `if (isFoo(result))` in the caller function.